10x/benchmark/10x_cost_analysis.ipynb

```diff
@@ -0,0 +1,295 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "import os\n",
+    "import json\n",
+    "from datetime import datetime"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook expects a single environment variable, `METADATA`, to be set to the location of the metadata file you want processed. If running interactively, uncomment the below cell to set this parameter within the notebook environment. \n",
+    "\n",
+    "This notebook will create three plots: \n",
+    "\n",
+    "1. an unmodified cost file (not saved by default)\n",
+    "2. cost per task, merging split/main/join tasks with the same prefix\n",
+    "3. run time per task, merging split/main/join tasks with the same prefix\n",
+    "\n",
+    "In addition, it will print the total run time in wall clock hours and cpu hours, as well as the total cost across all tasks. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# to run interactively, uncomment the line below\n",
+    "# os.environ['METADATA] = 'path_to_metadata.json'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "if 'METADATA' not in os.environ:\n",
+    "    raise RuntimeError('METADATA variable not set')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "100  9576  100  9576    0     0  37274      0 --:--:-- --:--:-- --:--:-- 37406\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "# download cost script\n",
+    "curl -o ${TMPDIR}get_cost.py https://raw.githubusercontent.com/ambrosejcarr/cromwell-manager/master/src/accessories/calculate_workflow_cost.py 2>&1 > /dev/null\n",
+    "\n",
+    "# run the cost script on the metadata\n",
+    "python2.7 ${TMPDIR}/get_cost.py -m ${METADATA} > ./cost_data.txt"
```

Comment:
I'm not sure if we need to convert python2.7 to python3 for get_cost.py later in the future?

Reply:
Sorry, it seems this get_cost.py is downloaded from another repo of yours. So it won't be an issue as long as it works.

Reply:
Correct; it's a command line utility being called by your local environment's python2.7. I put that script into my repo because I was getting authentication errors pulling it from the original dsde repo.

Reply:
We already have a copy of the cost script in skylab here: 
https://github.com/HumanCellAtlas/skylab/blob/master/smartseq2_single_sample/run_multi_samples/scripts/calculate_cost_workflow.py

Could you please use that one for now? We have a ticket to put that script in cromwell-tools, so later we can reference it from there.